ebu_tt_live/node/deduplicator.py

```diff
@@ -53,43 +56,30 @@ def process_document(self, document, **kwargs):
                 document.validate()
                 self.producer_carriage.emit_data(data=document, **kwargs)
 
-    def remove_duplication(self, document, original_styles, new_style_list, styling_element, region_element, span_style_id, region_style_id):
+    def remove_duplication(self, document, original_styles, original_regions, new_style_list, new_region_list, styling_element, region_element, span_style_id, region_style_id):
 
-        for style in document.tt.head.styling:
-            
+        for style in enumerate(document.tt.head.styling):
+            original_styles.append(style)
 
+        # for region in enumerate(document.tt.head.layout):
+        #     original_regions.append(region)
 
-        original_styles =   [
-                                bindings.style_type(   id='SEQ58.defaultStyle1',
-                                                        color='rgb(255,255,255)',
-                                                        backgroundColor='rgb(0,0,0)'),
+        mirror_styles = original_styles
 
-                                bindings.style_type(   id='SEQ59.defaultStyle1',
-                                                        color='rgb(255,255,255)',
-                                                        backgroundColor='rgb(0,0,0)')
+        mirror_styles_no_id = set()
 
-                                # [bindings.style_type(   id= 'SEQ60.defaultStyle1',
-                                #                         color= 'rgb(0,255,255)',
-                                #                         backgroundColor= 'rgb(0,0,0)')]
-                            ]
+        for mirror_entry in mirror_styles:
+            mirror_entry_no_id = mirror_entry.rstrip(style.id)
 
-        list_a = [[],[]]
-        list_b = [[],[]]
-        new_style_list = [[],[],[],[]]
+            mirror_styles_no_id.add(mirror_entry_no_id)
 
-        i = 0
+            mirror_styles_new_id = list(mirror_styles_no_id)
 
-        list_a[0].append(original_styles[i].color)
-        list_a[1].append(original_styles[i].backgroundColor)
+            for x in enumerate(mirror_styles_new_id):
+                new_style = mirror_styles_new_id.append("xml:id=\"style\"" + str(x))
 
-        for x in range(len(original_styles)):
-            list_b[0].append(original_styles[x].color)
-            list_b[1].append(original_styles[x].backgroundColor)
+                for span_style in enumerate(document.tt.body):
```

Comment:
I don't think `document.tt.body` will contain _any_ span styles - if you want to do a hierarchical depth first search you'll need to change this code quite a bit to use recursive iteration or to use the RecursiveOperation class.